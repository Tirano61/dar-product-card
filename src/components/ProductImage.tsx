import React from "react";
import { useContext } from "react";
import { ProductContext } from "./ProductCard";

import styles from "../styles/styles.module.css";
import noImage from "../assets/no-image.jpg";

interface Props {
  img?: string,
  className?: string
  style?: React.CSSProperties
}

export const ProductImage = ({ img = '', className, style }: Props) => {
  const { product } = useContext(ProductContext);
  let imgShow: string;
  if (img) {
    imgShow = img;
  } else if (product.img) {
    imgShow = product.img;
  } else {
    imgShow = noImage;
  }
  return (
    <img
      className={`${styles.productImg} ${className}`}
      style={style}
      src={imgShow}
      alt="Product image"
    />
  )
}