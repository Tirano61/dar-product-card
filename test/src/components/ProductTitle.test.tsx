
import React from "react";
import { render } from "@testing-library/react";
import { ProductTitle, ProductCard } from '../../../src/components';
import { product1 } from "../data/products";



describe('ProductTitle', () => {
  test('Debe mostrar el componente correctamente con el titulo personaliado', () => { 
      const wrapper = render(
        <ProductTitle title='Custon Title' className="custom-class" />
      )

      expect( wrapper.asFragment()).toMatchSnapshot();
      
  });
  test('Debe mostrar el conponente con el nombre del producto', () => { 
    const wrapper = render(
      <ProductCard product={ product1 } >
      {
        () =>(
          <ProductTitle />
        )
      }
      </ProductCard>
    )
    
    expect( wrapper.asFragment()).toMatchSnapshot();
  })
});