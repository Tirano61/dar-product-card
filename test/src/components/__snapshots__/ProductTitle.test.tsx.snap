// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle Debe mostrar el componente correctamente con el titulo personaliado 1`] = `
<DocumentFragment>
  <span
    class="productDescription custom-class"
  >
    Custon Title
  </span>
</DocumentFragment>
`;

exports[`ProductTitle Debe mostrar el conponente con el nombre del producto 1`] = `
<DocumentFragment>
  <div
    class="productCard undefined"
  >
    <span
      class="productDescription undefined"
    >
      Coffee - Mug - sin imagen
    </span>
  </div>
</DocumentFragment>
`;
